File: backend/app/schemas/team.py
from pydantic import BaseModel
from typing import Optional, List
from datetime import datetime

class MemberResponse(BaseModel):
    id: int
    name: str
    email: str
    role: str
    status: str = "offline"
    avatar: Optional[str] = None
    pipelines: int = 0
    joined_at: Optional[datetime] = None

    class Config:
        from_attributes = True

class ShareRequest(BaseModel):
    pipeline_id: int
    member_ids: List[int]
    permission: str = "view"  # view, edit, admin

class ShareResponse(BaseModel):
    id: int
    pipeline_id: int
    shared_by: int
    shared_with: List[int] = []
    permission: str
    shared_at: Optional[datetime] = None

class CommentCreate(BaseModel):
    pipeline_id: int
    content: str

class CommentResponse(BaseModel):
    id: int
    pipeline_id: int
    author_id: int
    author_name: Optional[str] = None
    content: str
    created_at: Optional[datetime] = None
    resolved: bool = False

    class Config:
        from_attributes = True

class InviteRequest(BaseModel):
    email: str
    role: str = "member"

File: frontend/src/types/team.ts
export interface TeamMember {
  id: number;
  name: string;
  email: string;
  role: string;
  status: 'online' | 'away' | 'offline';
  avatar: string;
  pipelines: number;
  joinedAt: string;
}

export interface Comment {
  id: number;
  pipelineId: number;
  author: TeamMember;
  content: string;
  createdAt: string;
  resolved: boolean;
  replies?: Comment[];
}

export interface PipelineVersion {
  id: number;
  pipelineId: number;
  version: string;
  author: TeamMember;
  changes: string;
  createdAt: string;
  code?: string;
}

export interface SharedPipeline {
  id: number;
  pipelineId: number;
  sharedBy: TeamMember;
  sharedWith: TeamMember[];
  permission: 'view' | 'edit' | 'admin';
  sharedAt: string;
}

export interface TeamActivity {
  id: number;
  user: TeamMember;
  action: string;
  target: string;
  time: string;
  type: 'success' | 'approval' | 'create' | 'query' | 'fix';
}
